### DISCUSSION TITLE
What is your impression?

### DISCUSSION THREAD

**Discussion by e1em3ntoDX** [DevExpress/BlazorReportDefinition#6] [Feedback]
Please share your thoughts on this implementation in this thread. We'd love to hear the pros and cons of creating reports this way compared to using the DevExpress Visual Studio Report Designer. If you see room for possible enhancements, share your ideas with us.

**Comment by costalino** [comment]
Hi

In my point of view the idea is good, but it's very difficult to use and not very user friendly compare to a report designer.

It will be create to work inside a designer that generate the razor code.

Regards
Lino

**Comment by skhchishti** [comment]
Hi,

This is good.

We can make it more simpler, if we can generate a report directly using select query from SQL or SQL Stored Procedures.
eg: select Col1 (Size 100), Col2(Size 200),Amount(70) from TableName where Col1=''

Can't we ignore the styling part, it is very tedious.

This will make development much easier.

Regards
Hadiuddin Chishti

**Comment by frankvandenbergh** [comment]
I started to use the report a couple of months ago. I use it to create lists of items (just like a grid and table in excel). And use very limited layout functionalities; just think about a title, grouping, summaries and a datetime at the bottom. Want as much of data on a screen as possible (limit white area's, but it should be readable). Your Table Report is neat, but I'd create a one page report.

Eg think of shipments. And then give a list of shipments where arrival times are missing (that someone has to track down manually).

I make the reports. First I ask what persons want to know, create the query, and use the wizard in VS. That works well, until someone wants a different content. I think that is cumbersome:
- I don't think about papersize and I've to alter that
- I have to do aligning of header and fields manually. Should be nice that column headers will "follow" details band fields.

What skhchishti suggesting is pretty much what I want. But I do like the styling.

Therefor I do think this is a nice approach. Where I can replace the Bands content dynamically (eg XControlLabel). This would create more consistant layouts. At least when I can create repx files (for our intranet).

**Reply by skhchishti** [reply]
Thanks Frank.

But if we can deliver it, it will be reduce our effort on development of reports to much extent, I can say more than 90%. Expecting a reply on the suggestion.

**Reply by e1em3ntoDX** [reply]
Thank you for your feedback. Did you have a chance to play with this API and create report documents? 

@skhchishti your suggestion a bit correlates with our past idea to introduce the Fluent API for table report creation. We postponed implementing this functionality the previous year ([see the 2021 roadmap](https://community.devexpress.com/blogs/reporting/archive/2021/02/16/devexpress-reporting-2021-roadmap.aspx)), but we might revise the priority for the current year. I'm afraid I'm not in a position to make any promises right now: we're currently in the planning phase and we'll definitely publish a roadmap once we create it. 

@frankvandenbergh can you please elaborate on the point related to the paper size? This will allow me to clearly understand the difficulties you face and see what we can do.

**Reply by frankvandenbergh** [reply]
Hi Dmitry,

I can't really use the example as there is no SaveLayoutToXml yet.

Regarding the paper size:
- Add class (eg XtraReport.cs), select Table Report from the Wizard
- Select a table with a lot of fields (say 15) for your report. If more you get a kind of "piano" in the designer
- At the end of the wizard you've to select a page. If I select A4 the fieldwidths are pretty small. So I change that to Custom and some width/height (what is best for a desktop screen?). So lets choose 400x400 mm.

Setting the width and height is a bit of trying. Or I have to calculate upfront how width each field will be, and do the math myself.

- Next try to insert a new field in the designer.
- drag/drop between two fields will replace a current field. Therefor I click Insert a column to the left. The header doesn't insert, so I do the same with the header. And drop the field in the detail.
- Next with Shift/Drag the end of the field to make it smaller.
- And I have to do that with the header text
- Next I have to make the pagenumber smaller and the group above
- Then I can make the "page" smaller

If I drag/drop a new field to the end of the detail fields, it will be inserted, but not aligned/grouped with the other fields. Even a different so it seems.

**Reply by e1em3ntoDX** [reply]
Hi Frank,

Thank you for the detailed update. I appreciate it. Am I correct that you only export reports and don't print them? I have never seen a paper with 400x400 mm dimensions.

>I can't really use the example as there is no SaveLayoutToXml yet.

Yes, this example only allows you to define a report structure in Razor markup syntax and preview it. I understand that testing this API will probably take time. So, it is not a problem if you didn't get the chance to test it yet. Thank you for all your feedback!

**Comment by frankvandenbergh** [comment]
Yes Dmitry,
I really try to make it a paperless office. I heard the term about 50 years ago, but still printers are making overtime.

It is always a struggle to give as much data/information with the little space there is. And A4 is much smaller than a screen. And usually people export the data to Excel to make notes, analysis, storage,...

BTW yesterday I received a policy document from the municipality and they were able to put so much text in the A4 that no one can actually read it. Only on a computer after zooming in. Perhaps deliberately.
